doc/conf.py

```diff
@@ -18,10 +18,31 @@
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
 import os
+import shutil
+import subprocess
 import sys
+import tempfile
+
 from qiskit import __version__
 sys.path.insert(0, os.path.abspath('.'))
 
+# Clone the tutorials-repo
+def clone_tutorials(app):
```

Comment:
Can we move this to a `make` target or similar? The general entry point for creating the documentation is the main `make doc`, which already contains a number of steps. It would fit more naturally into the current philosophy, as it is akin to some of the other steps that take care of "generating the files that will be rendered by sphinx" before invoking it and increases flexibility.

Reply:
Sure, I can do that. I wrote it this way to make the tutorial rendering independent of how we initiate a doc build, like manually running sphinx-build (which is how I tested this). Since the tutorials are likely going to serve as useful documentation I assumed we always wanted it there by default. But if you're saying that `make doc` is the only interface we support for building docs then I'll move it.

Reply:
Thanks - it was more about catering to the common usage `make doc` _and_ other scenarios! I'll expand in other comments as I think with some extra tweaks would be even cooler!